applications/main/nfc/scenes/nfc_scene_mf_classic_dict_attack.c

```diff
@@ -29,7 +34,9 @@ NfcCommand nfc_dict_attack_worker_callback(NfcGenericEvent event, void* context)
     } else if(mfc_event->type == MfClassicPollerEventTypeRequestMode) {
         const MfClassicData* mfc_data =
             nfc_device_get_data(instance->nfc_device, NfcProtocolMfClassic);
-        mfc_event->data->poller_mode.mode = MfClassicPollerModeDictAttack;
+        mfc_event->data->poller_mode.mode = (instance->nfc_dict_context.enhanced_dict) ?
```

Comment:
I see one place where this `enhanced_dict` is set to `true` on line 175, which means that we will start from `MfClassicPollerModeDictAttackEnhanced`. And how `MfClassicPollerModeDictAttackStandard` will be performed now? I'm asking because the only place where `enhanced_dict` is set to `false` is on line 385, in `_on_exit` callback.

Reply:
This flag is to indicate whether nested attacks should be attempted. Nested attacks are disabled during the CUID dictionary attack to force the default behavior (`MfClassicPollerModeDictAttackStandard`), as all of the keys in the CUID dictionary match the properties that the enhanced attack searches for. After `enhanced_dict` is set to `true`, it should no longer be set to `false` except upon exit as the CUID dictionary is tested first - the only time it is needed. The variable is necessary to inform the poller how it should behave during initial phase of the dictionary attack. Standard attacks are still attempted in `MfClassicPollerModeDictAttackEnhanced`, up until the point the first key is found.

Reply:
@RebornedBrain don't forget to resolve all conversations